src/marqo/tensor_search/tensor_search.py

```diff
@@ -1238,3 +1239,22 @@ def _get_model_properties(index_info):
                 f"Please provide model_properties if the model is a custom model and is not supported by default")
 
     return model_properties
+
+def get_loaded_models() -> dict:
+    available_models = s2_inference.get_available_models()
+    message = {
+        'models' : [
+            {"model_name": ix} for ix in available_models
+        ]
+    }
+    return message
+def eject_model(model_name: str, device: str) -> dict:
+    try:
+       result = s2_inference.eject_model(model_name, device)
+    except s2_inference_errors.ModelNotLoadedError as e:
+        raise errors.ModelNotLoadedError(message=str(e))
+    return result
+def get_cuda_info() -> dict:
+    return {"device": "cuda",
+            "memory_usage": f"{round(torch.cuda.memory_allocated() / 1024**3, 1)} GiB",
+            "total_device_memory": f"{round(torch.cuda.get_device_properties(0).total_memory/ 1024**3, 1)} GiB"}
```

Comment:
The cuda functions should be moved to a separate file. I am thinking 
`s2_inference/devices/cuda.py`. We want to remain open to other designs, such as an inference server. 

Thoughts @jn2clark?

Reply:
leave for future work